client/connection/manager.go

```diff
@@ -55,33 +68,72 @@ func (manager *connectionManager) Connect(consumerID, providerID identity.Identi
 		}
 	}()
 
-	proposal, err := manager.findProposalByProviderID(providerID)
+	cancelable := utils.NewCancelable()
+	manager.closeAction = utils.CallOnce(func() {
+		log.Info(managerLogPrefix, "Closing active connection")
+		manager.status = statusDisconnecting()
+		cancelable.Cancel()
+	})
+
+	val, err := cancelable.
+		Request(func() (interface{}, error) {
+			return manager.findProposalByProviderID(providerID)
+		}).
+		Call()
```

Comment:
Looks very unusual to keep `.` in the previous line - is it a common practice in Go?
I wasn't able to find these usages because I was looking for `.Call()` :D

Reply:
Yes that go syntax style. Otherwise - syntax error :)